include/rocksdb/perf_context.h

```diff
@@ -168,6 +188,8 @@ struct PerfContext {
   uint64_t env_lock_file_nanos;
   uint64_t env_unlock_file_nanos;
   uint64_t env_new_logger_nanos;
+  std::map<uint32_t, PerfContextByLevel>* level_to_perf_context;
```

Comment:
I think about it more. Not just PerfContextByLevel, but the map itself is leaked after the thread exits. Does C++ allow a constructor of PerfContext? If it allows, we should destruct the map. If not, we should clearly comment in the EnablePerLevelPerfContext() that the user is responsible to call ClearPerLevelPerfContext() before the thread exits. Otherwise, something will be leaked.

Reply:
I tried to add a destructor for PerfContext class but got compilation error:
> monitoring/perf_context.cc:18:22: error: non-local variable ‘rocksdb::perf_context’ declared ‘__thread’ has a non-trivial destructor
 __thread PerfContext perf_context;
monitoring/perf_context.cc:18:22: note: C++11 ‘thread_local’ allows dynamic initialization and destruction

I will add comments to advise users to always pair `EnablePerLevelPerfContext()` with `ClearPerLevelPerfContext()`

Reply:
tested with `thread_local` and make check passed, I will replace __thread with thread_local in a separate PR